test/test_transforms_v2_refactored.py

```diff
@@ -860,3 +862,448 @@ def test_transform_noop(self, input_type, device):
+    @pytest.mark.parametrize(
+        ("input_type", "kernel"),
+        [
+            (torch.Tensor, F.affine_image_tensor),
+            (PIL.Image.Image, F.affine_image_pil),
+            (datapoints.Image, F.affine_image_tensor),
+            (datapoints.BoundingBox, F.affine_bounding_box),
+            (datapoints.Mask, F.affine_mask),
+            (datapoints.Video, F.affine_video),
+        ],
+    )
+    def test_dispatcher_signature(self, kernel, input_type):
+        check_dispatcher_signatures_match(F.affine, kernel=kernel, input_type=input_type)
+
+    @pytest.mark.parametrize(
+        "input_type",
+        [torch.Tensor, PIL.Image.Image, datapoints.Image, datapoints.BoundingBox, datapoints.Mask, datapoints.Video],
+    )
+    @pytest.mark.parametrize("device", cpu_and_cuda())
+    def test_transform(self, input_type, device):
+        input = self._make_input(input_type, device=device)
+
+        check_transform(transforms.RandomAffine, input, **self._CORRECTNESS_TRANSFORM_AFFINE_RANGES)
+
+    @pytest.mark.parametrize("angle", _CORRECTNESS_AFFINE_KWARGS["angle"])
+    @pytest.mark.parametrize("translate", _CORRECTNESS_AFFINE_KWARGS["translate"])
+    @pytest.mark.parametrize("scale", _CORRECTNESS_AFFINE_KWARGS["scale"])
+    @pytest.mark.parametrize("shear", _CORRECTNESS_AFFINE_KWARGS["shear"])
+    @pytest.mark.parametrize("center", _CORRECTNESS_AFFINE_KWARGS["center"])
+    @pytest.mark.parametrize(
+        "interpolation", [transforms.InterpolationMode.NEAREST, transforms.InterpolationMode.BILINEAR]
+    )
+    @pytest.mark.parametrize("fill", _CORRECTNESS_FILL)
+    def test_functional_image_correctness(self, angle, translate, scale, shear, center, interpolation, fill):
+        image = self._make_input(torch.Tensor, dtype=torch.uint8, device="cpu")
+
+        fill = self._adapt_fill(fill, dtype=torch.uint8)
+
+        actual = F.affine(
+            image,
+            angle=angle,
+            translate=translate,
+            scale=scale,
+            shear=shear,
+            center=center,
+            interpolation=interpolation,
+            fill=fill,
+        )
+        expected = F.to_image_tensor(
+            F.affine(
+                F.to_image_pil(image),
+                angle=angle,
+                translate=translate,
+                scale=scale,
+                shear=shear,
+                center=center,
+                interpolation=interpolation,
+                fill=fill,
+            )
+        )
+
+        mae = (actual.float() - expected.float()).abs().mean()
+        assert mae < 5
+
+    @pytest.mark.parametrize("center", _CORRECTNESS_AFFINE_KWARGS["center"])
+    @pytest.mark.parametrize(
+        "interpolation", [transforms.InterpolationMode.NEAREST, transforms.InterpolationMode.BILINEAR]
+    )
+    @pytest.mark.parametrize("fill", _CORRECTNESS_FILL)
+    @pytest.mark.parametrize("seed", list(range(10)))
+    def test_transform_image_correctness(self, center, interpolation, fill, seed):
```

Comment:
I guess we can't unify this test with the functional one, because the parameters are too different (i.e. they're randomly sampled in the transform)?

That being said I'm curious why the MAE threshold are different in both test

Reply:
> I guess we can't unify this test with the functional one, because the parameters are too different (i.e. they're randomly sampled in the transform)?

Correct.

> That being said I'm curious why the MAE threshold are different in both test

I just put the minimal needed threshold for the tests to pass. I guess in our case transform test sampled other affine params that ultimately led to a higher MAE. I'll unify them.